src/main/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDump.java

```diff
@@ -94,11 +99,27 @@ public HttpResponse doActivate(@QueryParameter String opts) throws Exception {
 
         int exitCode = p.waitFor();
 
-        if (exitCode!=0)
-            // As the real cause of the failure of the agent is lost and we cannot access it, we can only recommend to
-            // see the logs.
-            throw new Error("Failed to activate file leak detector.\nPerhaps wrong parameters.\n" +
-                    "See logs for more info.\nSpecifically, look for 'Agent failed to start!' or something related\n\n");
+        if (exitCode!=0) {
+            /*
+            There are 2 high-level ways the process can fail:
+            1. The process can fail in org.kohsuke.file_leak_detector.Main#main. For example, this can happen if Jenkins
+               is running with a JRE instead of a full JDK and so the instrumentation API doesn't load. In this case,
+               the output of the process we created here contains the root cause of the failure. Because it is a
+               separate process, its output will _not_ be sent to Jenkins' logs/stdout/stderr by default, and so we
+               _must_ include its output in this error message or that information will be lost forever.
+            2. The process can fail in org.kohsuke.file_leak_detector.AgentMain#agentmain. For example, this can happen
+               if an invalid option is passed to the agent. In this case, the root cause of the issue will be printed
+               out to the stderr of the Jenkins process, and so the user needs to look there to find out  what went wrong.
+            We use Failure to omit the stack trace when it is shown to the user. Otherwise, the exception gets wrapped in
+            ServletException, and the message is duplicated, which is confusing when we have such a large message like this.
+            */
+            Exception e = new Failure("Failed to activate file leak detector. Perhaps the parameters were incorrect. " +
+                    "Look for 'Agent failed to start!' in stderr logs for more info. Additional logs:\n"+baos, true);
```

Comment:
Findbugs doesn't like the explicit call to `baos.toString()` because it relies on the default charset, but in this case, we are reading stdout/stderr, which is written using the default charset, so reading it with the default charset is correct. I think we could also be explicit to make Findbugs happy and use `baos.toString(Charset.defaultCharset().name())`, but the current code seemed simpler.